Beginnings of the branching section

georgebrock opened the pull request:
Adds chapters on the following subjects:
- Intro to branches
- Create branches
- Compare branches
- Create aliases
- Merge branches

This is roughly the first half of the branching section.

georgebrock committed:
Expand "Refer to commits" to include HEAD.

georgebrock committed:
Draft branching content: intro, create, compare

The introduction outlines what branching is, and what it is for.

The chapter on creation outlines how to make a branch, and how to switch
between branches.

* git branch
* git checkout <branch>
* git checkout -b <branch>

The chapter on comparison outlines how to compare branches.

* git log --graph --decorate --all
* git show-branch <first-branch> <second-branch> ...

georgebrock committed:
Add examples to branch creation.

georgebrock committed:
Create commits with content.

georgebrock committed:
Add examples to compare branches chapter.

georgebrock committed:
Tweak branching introduction.

georgebrock committed:
Insert a stub chapter on aliases.

georgebrock committed:
Aside: creating aliases.

georgebrock committed:
Add draft chapter on merging branches.

georgebrock committed:
Clarify explanation of branches.

calebhearth commented on src/2-2-refer-to-commits.md:
```diff
@@ -54,6 +54,23 @@ the parent commit:
+As we build up our project, it's often useful to be able to refer to "the commit
+I just made".
+
+Git uses the name `HEAD` to refer to the current revision of the project, which
```
`@` is an alias for `HEAD` as well.

georgebrock committed:
Use new example style.

Since this branch was created, the way examples are handled has changed
quite a bit.

This commit updates the existing examples in chapter 3 to use the new style,
and adds support for setting up an example from a remote with multiple
branches.
